doc/source/api.rst

```diff
@@ -1627,6 +1627,52 @@ Conversion
 
 .. currentmodule:: pandas
 
+PeriodIndex
+--------------
+
+.. autosummary::
+   :toctree: generated/
+
+   PeriodIndex
```

Comment:
I think we also want to have this as 'class without autosummary' to not include all the attributes (but listing the different ones as you did below is certainly fine!)

Reply:
Apart from adding the "class_without_autosummary" template, you should also add it to the 'hack' in numpydoc, see the diff of https://github.com/pandas-dev/pandas/pull/17642

Reply:
Ok, done (I think, not sure what these actually do, please review it it's as intended).

Reply:
yes that looks good